### PR TITLE
feat(argo-rollouts): Allow defining priorityClassName

### LINKED ISSUE #1211: Ability to provide priorityClassName to argo-rollouts controller
### Is your feature request related to a problem?

Due to its importance, we would like to be able to give a high priority to the controller. Unfortunately, the chart doesn't currently support that.

### Related helm chart

argo-rollouts

### Describe the solution you'd like

Something like this should be added to the [deployment.yaml template](https://github.com/argoproj/argo-helm/blob/master/charts/argo-rollouts/templates/controller/deployment.yaml):
```yaml
{{- with .Values.controller.priorityClassName }}
  priorityClassName: {{ . }}
{{- end }}
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amirschw**
Add the ability to define priorityClassName for the argo-rollouts controller and dashboard pods.

Fixes #1211.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

Signed-off-by: amirschw <24677563+amirschw@users.noreply.github.com>

### CODE REVIEW

**Review by mkilchhofer — COMMENTED** [review]
Hi @amirschw , thanks for your contribution. WDYT would it make sense to also add it for the dashboard component?

**Comment by amirschw** [comment]
That makes perfect sense, thanks @mkilchhofer. I will update the PR to include it.

File: charts/argo-rollouts/values.yaml [hunk 1]
```diff
@@ -144,6 +146,8 @@ dashboard:
   tolerations: []
   # -- Assign custom [affinity] rules to the deployment
   affinity: {}
+  # -- [priorityClassName] for the dashboard server
+  priorityClassName: priorityClassName
```
Comment (mkilchhofer) [inline charts/argo-rollouts/values.yaml:150]:
```suggestion
  priorityClassName: ""
```

Comment (amirschw) [reply charts/argo-rollouts/values.yaml]: whoops, good catch, thanks!

**Review by mkilchhofer — APPROVED** [review]
lgtm. thanks for your contribution

**Pull request merged**